storage/v1alpha1/types_share.go

```diff
@@ -0,0 +1,121 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SharedResource allows a backing Secret or ConfigMap to be shared across namespaces.
+// Pods can mount the shared Secret or ConfigMap by adding a CSI volume to the pod specification using the
+// "sharedresource.csi.storage.openshift.io" CSI driver and a reference to the SharedResource in the volume attributes:
+//
+// spec:
+//  volumes:
+//  - name: shared-secret
+//    csi:
+//      driver: sharedresource.csi.storage.openshift.io
+//      volumeAttributes:
+//        sharedResource: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to get the named SharedResource object with an appropriate
+// ClusterRole and ClusterRoleBinding.
```

Comment:
What is the RBAC permission required?

Reply:
Pushed update with added example of ClusterRole and ClusterRoleBinding